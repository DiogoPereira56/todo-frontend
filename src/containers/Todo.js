import TopNavbar from '../components/TopNavbar';


const Todo = () => {
  return (
      <TopNavbar />
  );
}

export default Todo;
