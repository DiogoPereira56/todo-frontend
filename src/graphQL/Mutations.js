import { gql } from '@apollo/client';

//User login and resgistration
export const LOGIN_USER_MUTATION = gql`
    mutation clientLogin($email: String!, $password: String!) {
        login(email: $email, password: $password)
    }
`;

export const REGISTER_USER_MUTATION = gql`
    mutation clientRegister($email: String!, $name: String!, $password: String!) {
        register(name: $name, email: $email, password: $password)
    }
`;

export const CLIENT_LISTS_MUTATION = gql`
    mutation getClient($limit: Float!, $offset: Float!) {
        getClientInformations {
            name
            idClient
            list(limit: $limit, offset: $offset) {
                idList
                idClient
                listName
            }
        }
    }
`;

//List actions
export const NEW_LIST_MUTATION = gql`
    mutation addNewList($listName: String!) {
        addList(listName: $listName)
    }
`;

export const DELETE_LIST_MUTATION = gql`
    mutation($idList: Float!, $idClient: Float!) {
        removeList(idList: $idList, idClient: $idClient)
    }
`;

export const RENAME_LIST_MUTATION = gql`
    mutation newList(
        $idList: Float!
        $title: String!
        $idClient: Float!
        $limit: Float!
        $offset: Float!
        $orderByTitle: Boolean!
        $order: String!
    ) {
        updateList(idList: $idList, title: $title, idClient: $idClient) {
            idList
            listName
            taskss(limit: $limit, offset: $offset, orderByTitle: $orderByTitle, order: $order) {
                idTask
                title
                complete
                description
            }
        }
    }
`;

export const LIST_INFO_MUTATION = gql`
    mutation getListInfo(
        $idList: Float!
        $idClient: Float!
        $limit: Float!
        $offset: Float!
        $orderByTitle: Boolean!
        $order: String!
    ) {
        getList(idList: $idList, idClient: $idClient) {
            idList
            idClient
            listName
            taskss(limit: $limit, offset: $offset, orderByTitle: $orderByTitle, order: $order) {
                idTask
                title
                complete
                description
            }
        }
    }
`;

export const GET_CLIENT_TOTAL_LISTS = gql`
    mutation {
        getClientTotalLists
    }
`;

//Task Actions
export const NEW_TASK_MUTATION = gql`
    mutation newTask($title: String!, $idList: Float!, $idClient: Float!) {
        addTask(title: $title, idList: $idList, idClient: $idClient) {
            idTask
            idList
            title
            complete
            description
        }
    }
`;

export const DELETE_TASK_MUTATION = gql`
    mutation($idTask: Float!, $idClient: Float!) {
        removeTask(idTask: $idTask, idClient: $idClient)
    }
`;

export const UPDATE_TASK_DESCRIPTION_MUTATION = gql`
    mutation($idTask: Float!, $description: String!, $idClient: Float!) {
        updateTaskDescription(idTask: $idTask, description: $description, idClient: $idClient) {
            idTask
            title
            complete
            description
        }
    }
`;

export const UPDATE_TASK_TITLE_MUTATION = gql`
    mutation($idTask: Float!, $title: String!, $idClient: Float!) {
        updateTaskTitle(idTask: $idTask, title: $title, idClient: $idClient) {
            idTask
            title
            complete
            description
        }
    }
`;

export const UPDATE_TASK_COMPLETION_MUTATION = gql`
    mutation($idTask: Float!, $complete: Boolean!, $idClient: Float!) {
        updateTaskCompletion(idTask: $idTask, complete: $complete, idClient: $idClient) {
            idTask
            idList
            title
            complete
            description
        }
    }
`;

export const LIST_TOTAL_TASK_MUTATION = gql`
    mutation listTotalTasks($idList: Float!, $idClient: Float!) {
        getListsTotalTasks(idList: $idList, idClient: $idClient)
    }
`;

export const ALL_CLIENT_TASKS_MUTATION = gql`
    mutation($limit: Float!, $offset: Float!, $idClient: Float!, $orderByTitle: Boolean!, $order: String!) {
        getAllTasks(
            limit: $limit
            offset: $offset
            idClient: $idClient
            orderByTitle: $orderByTitle
            order: $order
        ) {
            idTask
            idList
            title
            complete
            description
        }
    }
`;
export const SEARCHED_TASKS_MUTATION = gql`
    mutation(
        $limit: Float!
        $offset: Float!
        $idClient: Float!
        $orderByTitle: Boolean!
        $search: String!
        $order: String!
    ) {
        getSearchedTasks(
            limit: $limit
            offset: $offset
            idClient: $idClient
            orderByTitle: $orderByTitle
            search: $search
            order: $order
        ) {
            idTask
            idList
            title
            complete
            description
        }
    }
`;
export const TOTAL_SEARCHED_TASKS_MUTATION = gql`
    mutation($idClient: Float!, $search: String!) {
        getTotalSearchedTasks(idClient: $idClient, search: $search)
    }
`;
