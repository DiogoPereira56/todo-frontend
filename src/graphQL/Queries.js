import {gql} from '@apollo/client'


export const DECODED_TOKEN = gql`
  {
    getDecodedToken{
      id
      name
    }
  }
`

export const LOG_OUT = gql`
  {
    removeToken
  }
`

export const GET_ALL_LISTS = gql`
  {
    lists{
      idList
      idClient
      listName
    }
  }
`

export const GET_CLIENT_INFORMATION = gql`
{
  getClientInformation{
    name
    lists{
      idList
      listName
      tasks{
        idTask
        title
        complete
        description
      }
    }
  }
}
`
