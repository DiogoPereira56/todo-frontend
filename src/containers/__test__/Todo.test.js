/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import Todo from '../Todo';
import { GET_CLIENT, CLIENT_TOTAL_LISTS, GET_LIST_TASKS } from '../../graphQL/Queries';
import ReactDOM from 'react-dom';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';

const mockGetClientData = () => {
    return [
        {
            request: {
                query: GET_CLIENT,
                variables: {
                    limit: 10,
                    offset: 0,
                },
            },
            result: {
                data: {
                    getClientInformation: {
                        idClient: 3,
                        name: 'Diogo Pereira',
                        __typename: 'Client',
                        list: [
                            { idList: 1, listName: 'lista', idClient: 3, __typename: 'ListOfTasks' },
                            { idList: 2, listName: 'compras', idClient: 3, __typename: 'ListOfTasks' },
                        ],
                    },
                },
            },
        },
        {
            request: {
                query: CLIENT_TOTAL_LISTS,
            },
            result: {
                data: { getTotalLists: 2 },
            },
        },
        {
            request: {
                query: GET_LIST_TASKS,
                variables: {
                    idList: 2,
                    idClient: 3,
                    limit: 2,
                    offset: 0,
                    orderByTitle: false,
                    order: 'ASC',
                },
            },
            result: {
                data: {
                    listQuery: {
                        idList: 2,
                        idClient: 3,
                        listName: 'compras',
                        __typename: 'ListOfTasks',
                        taskss: {
                            tasks: [
                                {
                                    idTask: 4,
                                    title: 'pao',
                                    complete: true,
                                    description: 'falta',
                                    __typename: 'task',
                                },
                                {
                                    idTask: 20,
                                    title: 'arroz',
                                    complete: true,
                                    description: '',
                                    __typename: 'task',
                                },
                            ],
                            hasMore: 33,
                            __typename: 'TasksResolveField',
                        },
                    },
                },
            },
        },
    ];
};

describe('<Todo />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MockedProvider>
                <Todo></Todo>
            </MockedProvider>,
            div,
        );
    });

    it('matches Todo loading snapshot', () => {
        const tree = renderer
            .create(
                <MockedProvider>
                    <Todo></Todo>
                </MockedProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('matches Todo success snapshot', async () => {
        const mockedClientData = mockGetClientData();
        const tree = renderer
            .create(
                <MockedProvider mocks={mockedClientData} addTypename={false}>
                    <Todo></Todo>
                </MockedProvider>,
            )
            .toJSON();
        await new Promise((resolve) => setTimeout(resolve, 0));
        expect(tree).toMatchSnapshot();
    });

    it('renders todo correctly', async () => {
        const mockedClientData = mockGetClientData();
        const { getByTestId } = render(
            <MockedProvider mocks={mockedClientData}>
                <Todo></Todo>
            </MockedProvider>,
        );
        await new Promise((resolve) => setTimeout(resolve, 0));
        screen.debug();

        /* const wrapper = await waitFor(() => getByTestId('wrapper'));
        console.log(prettyDOM(wrapper)); */

        //expect(1).toEqual(2);
    });
});
