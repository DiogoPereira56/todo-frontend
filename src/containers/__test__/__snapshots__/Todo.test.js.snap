// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> matches Todo loading snapshot 1`] = `
<p>
  Loading...
</p>
`;

exports[`<Todo /> matches Todo success snapshot 1`] = `
<p>
  Loading...
</p>
`;
